internal/services/batch/batch_pool.go

```diff
@@ -868,11 +869,17 @@ func expandBatchPoolExtension(ref map[string]interface{}) (*pool.VmExtension, er
 	}
 
 	if settings, ok := ref["settings_json"]; ok {
-		result.Settings = pointer.To(settings)
+		err := json.Unmarshal([]byte(settings.(string)), &result.Settings)
+		if err != nil {
+			return nil, err
```

Comment:
Can we wrap this error for a bit more info
```suggestion
			return nil, fmt.Errorf("unmarshaling `settings_json`: %+v", err)
```